server/controllers/DocumentController.js

```diff
@@ -0,0 +1,183 @@
+            error,
+            message: 'Something went wrong! The document could not be saved.'
+          }));
+      })
+      .catch(error => response.status(500).send({
+        error,
+        message: 'The document could not be saved.'
+      }));
+  },
+  delete(request, response) {
+    return Documents
+      .findById(parseInt(request.params.id, 10), {})
+      .then((document) => {
+        if (!document) {
+          return response.status(400).send({
+            message: 'Document not found'
+          });
+        }
+        return document
+          .destroy()
+          .then(() =>
+            response.status(200).send({
+              message: 'Document deleted successfully.'
+            })
+          )
+          .catch(error => response.status(500).send({
+            error,
+            message: 'Something went wrong! The document could not be saved.'
+          }));
+      })
+      .catch(error => response.status(500).send({
+        error,
+        message: 'The document could not be deleted.'
+      }));
+  },
+  getAll(request, response) {
+    Documents
+      .findAll()
+      .then(documents => response.status(200).send(documents))
+      .catch(error => response.status(400).send(error));
+  },
+  getMyDocuments(req, res) {
+    return Documents
+    .findAll({
+      where: {
+        $or: [
+          { viewAccess: 'public' },
+          {
+            role: String(req.decoded.data.roleId)
+          },
+          {
+            userId: req.params.id
+          }
+        ]
+      },
+      include: [Users],
+      order: [['updatedAt', 'DESC']]
+    })
+    .then((document) => {
+      if (!document) {
+        return res.status(404).send({
+          message: 'Document Not Found'
+        });
+      }
+      return res.status(200).send(document);
+    })
+    .catch(error => res.status(400).send({
+      error,
+      message: 'Error occurred while retrieving documents'
+    }));
+  },
+  getOne(request, response) {
+    return Documents
+      .findById(parseInt(request.params.id, 10), {})
+      .then((document) => {
+        if (!document) {
+          return response.status(404).send({
+            message: 'Document does not exist'
+          });
+        }
+        return response.status(200).send({
+          document,
+          message: 'Document loaded'
+        });
+      })
+      .catch(error => response.status(500).send({
+        error,
+        message: 'Something went wrong! The document could not be opened.'
+      }));
+  },
+  search(request, response) {
+    const search = request.query.search;
+    return Documents
+      .findAll({
+        where: {
+          $or: [{
+            title: {
+              $iLike: `%${search}%`
+            }
+          }, {
+            content: {
+              $iLike: `%${search}%`
+            }
+          }]
+        }
+      })
+      .then((documents) => {
+        if (documents.length <= 0) {
+          return response.status(404)
```

Comment:
I stated this issue above, you should go through your codebase and try refactoring it